flang/lib/Lower/Bridge.cpp

```diff
@@ -1740,20 +1792,51 @@ class FirConverter : public Fortran::lower::AbstractConverter {
               },
               linkage);
         }
-      } else {
-        global = builder->createGlobal(loc, genType(var), globalName, linkage);
       }
-      global.setVisibility(mlir::SymbolTable::Visibility::Public);
-      auto addrOf = builder->create<fir::AddrOfOp>(loc, global.resultType(),
-                                                   global.getSymbol());
-      Fortran::lower::StatementContext stmtCtx;
-      mapSymbolAttributes(var, storeMap, stmtCtx, addrOf);
     } else if (sym.has<Fortran::semantics::CommonBlockDetails>()) {
-      llvm_unreachable("COMMON symbol processed elsewhere");
+      mlir::emitError(loc, "COMMON symbol processed elsewhere");
     } else {
-      TODO("global"); // Procedure pointer or something else
+      TODO("Lowering global procedure pointer definition");
```

Comment:
I think the "or something else" could previously be hit by module variables used in a function (they have `HostAssocDetails`, not ObjectEntityDetails). I will leave the previous comment because something similar may happen to contained procedure right now, but after that, I only see global procedure pointer that would be left here.